neo/rawio/neuralynxrawio/neuralynxrawio.py

```diff
@@ -189,15 +189,19 @@ def _parse_header(self):
             for idx, chan_id in enumerate(chan_ids):
                 chan_name = chan_names[idx]
 
-                chan_uid = (chan_name, chan_id)
+                chan_uid = (chan_name, str(chan_id))
                 if ext == 'ncs':
+                    if info['sampling_rate'] not in ncs_sampling_rates:
+                        ncs_sampling_rates.append(info['sampling_rate'])
+                        stream_id += 1
+
```

Comment:
I updated the stream handling to take into account the sampling rate, number of packets and start time of the signal. The assignment of the stream id still depends on the order in which the ncs files are being read. This implies that when loading a different subset of of the files, the stream_id is not necessarily preserved. Not sure how to prevent this since there is no way to guess how many file the complete session might have had in the first place...